crates/dreammaker/src/objtree.rs

```diff
@@ -266,6 +266,17 @@ impl<'a> TypeRef<'a> {
         }
     }
 
+    /// Recursively visit this and all child **types**.
+    pub fn recurse_types<F: FnMut(TypeRef<'a>)>(&self, f: &mut F) {
+        self.recurse(f);
+        for parent_typed_idx in &self.tree.redirected_parent_types {
```

Comment:
this is going to visit things multiple times

Reply:
Is it? `redirected_parent_types` should only be populated with things that have different `parent_type`s from their default path. So if `recurse` only hits direct child paths I don't see how it'd hit things multiple times.